reference/conanfile/methods/build_requirements.rst

```diff
@@ -0,0 +1,52 @@
+.. _reference_conanfile_methods_build_requirements:
+
+build_requirements()
```

Comment:
Somewhere we should link: https://docs.conan.io/en/2.0/tutorial/consuming_packages/use_tools_as_conan_packages.html
https://docs.conan.io/en/2.0/tutorial/creating_packages/other_types_of_packages/tool_requires_packages.html
And link back to here